src/develop/develop.c

```diff
@@ -1553,11 +1553,16 @@ int dt_dev_distort_backtransform(dt_develop_t *dev, float *points, size_t points
 int dt_dev_distort_transform_plus(dt_develop_t *dev, dt_dev_pixelpipe_t *pipe, int pmin, int pmax,
                                   float *points, size_t points_count)
 {
+  dt_pthread_mutex_lock(&dev->history_mutex);
```

Comment:
i think it's exactly what he means there, locking the history mutex.
usually we make sure you don't have to do this because you operate either
in gui thread on module-> or in process() thread on piece-> data. these
functions can unfortunately be called from both gui and process threads, so
you need to take care the history doesn't change under your feet, i.e.
commit_params() isn't called while you evaluate.

the alternative is providing two such functions to be called from the two
different thread realms, and operating on different input data. as far as i
understand though some of this input data is only generated in
commit_params or modify_roi_in/out? so a double implementation would get
messy and cause some duplicated code, potentially inside all distortion
modules.

On Tue, Jan 6, 2015 at 4:17 AM, Roman Lebedev <notifications@github.com>
wrote:

> In src/develop/develop.c
> <https://github.com/darktable-org/darktable/pull/790#discussion-diff-22465606>
> :
>
> > @@ -1553,11 +1553,16 @@ int dt_dev_distort_backtransform(dt_develop_t *dev, float *points, size_t points
> >  int dt_dev_distort_transform_plus(dt_develop_t *dev, dt_dev_pixelpipe_t *pipe, int pmin, int pmax,
> >                                    float *points, size_t points_count)
> >  {
> > +  dt_pthread_mutex_lock(&dev->history_mutex);
>
> This looks like a dirty hack to me, i'm afraid there are more issues
> caused by same cause as this one.
> So i'd sat, deeper investigation is required...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/darktable-org/darktable/pull/790/files#r22465606>.
>